plugin/src/main/java/com/denizenscript/denizen/events/block/CrafterCraftsScriptEvent.java

```diff
@@ -0,0 +1,91 @@
+package com.denizenscript.denizen.events.block;
+
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import org.bukkit.Keyed;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.CrafterCraftEvent;
+
+public class CrafterCraftsScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // crafter crafts <item>
+    //
+    // @Group Block
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a crafter crafts an item.
+    //
+    // @Context
+    // <context.location> returns the LocationTag of the crafter.
+    // <context.item> returns the ItemTag of the item being crafted.
+    // <context.recipe_id> returns the ElementTag of the recipe ID formed.
+    //
+    // @Determine
+    // ItemTag to set the item being crafted.
+    //
+    // -->
+
+    public CrafterCraftsScriptEvent() {
+        registerCouldMatcher("crafter crafts <item>");
+    }
+
+    public LocationTag location;
+    public ItemTag result;
+    public CrafterCraftEvent event;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        if (!runInCheck(path, location)) {
+            return false;
+        }
+        if  (!path.tryArgObject(2, result)) {
+            return false;
+        }
+        return super.matches(path);
+    }
+
+    @Override
+    public boolean applyDetermination(ScriptPath path, ObjectTag determinationObj) {
```

Comment:
this is the legacy determination handler, please use the modern registration method